### ISSUE TITLE
Collect count not working with sparse index

### ISSUE THREAD

**Issue by thierry-github** [arangodb/arangodb#19354]
## My Environment

* __ArangoDB Version__:        Community Edition 3.11.1
* __Deployment Mode__:       Single Server 
* __Deployment Strategy__:   ArangoDB Starter
* __Infrastructure__:               own
* __Operating System__:        Windows 10
* __Total RAM in your machine__:        32Gb
* __Disks in use__:        SSD

Hy,

I run this Query to get records (@path is an array of number)

```
FOR rec IN k_ref 
    FILTER 
        rec.__localScope IN  @path AND
        rec.__p_13 == 73
    RETURN rec
```

It works as expected : i have all my records (11)

Now adding collect to get the count (no changes to the query filters )

```
FOR rec IN k_ref 
    FILTER 
        rec.__localScope IN  @path AND
        rec.__p_13 == 73
    COLLECT WITH COUNT INTO countVariable
    return countVariable
```

Result : 0, 11 expected

This query uses an index (sparse) and all the fields are in the index.

```
Query String (166 chars, cacheable: true):
     FOR rec IN k_ref 
     FILTER 
         rec.__localScope IN  @path AND
         rec.__p_13 == 73
     COLLECT WITH COUNT INTO countVariable
     return countVariable


Execution plan:
 Id   NodeType        Est.   Comment
  1   SingletonNode      1   * ROOT
  7   IndexNode         51     - FOR rec IN k_ref   /* persistent index scan, scan only */    
  5   CollectNode        1       - COLLECT AGGREGATE countVariable = LENGTH()   /* count */
  6   ReturnNode         1       - RETURN countVariable

Indexes used:
 By   Name          Type         Collection   Unique   Sparse   Cache   Selectivity   Fields                         Stored values   Ranges
  7   referencial   persistent   k_ref        false    true     false        5.79 %   [ `__p_13`, `__localScope` ]   [  ]            ((rec.`__p_13` == 73) && (rec.`__localScope` IN [ 0, 1, 1007504, 1009247 ]))

Optimization rules applied:
 Id   RuleName
  1   use-indexes
  2   remove-filter-covered-by-index
  3   remove-unnecessary-calculations-2

43 rule(s) executed, 1 plan(s) created, peak mem [b]: 0, exec time [s]: 0.00026
```
If i remove the filter applied to the array (@path),  it now doesn't use the index and the collect works, and i get the right count (11).

```
Query String (127 chars, cacheable: true):
     FOR rec IN k_ref 
     FILTER 
         rec.__p_13 == 73
     COLLECT WITH COUNT INTO countVariable
     return countVariable

Execution plan:
 Id   NodeType                  Est.   Comment
  1   SingletonNode                1   * ROOT
  2   EnumerateCollectionNode    968     - FOR rec IN k_ref   /* full collection scan (projections: `__p_13`)  */   FILTER (rec.`__p_13` == 73)   /* early pruning */
  5   CollectNode                  1       - COLLECT AGGREGATE countVariable = LENGTH()   /* count */
  6   ReturnNode                   1       - RETURN countVariable

Indexes used:
 none

Optimization rules applied:
 Id   RuleName
  1   move-filters-into-enumerate
  2   reduce-extraction-to-projection

43 rule(s) executed, 1 plan(s) created, peak mem [b]: 0, exec time [s]: 0.00037
```

This query worked in the previous version 3.10.

Tks

**Comment by dheerajv** [comment]
We are seeing the same issue with persistent unique index.

**Comment by neunhoef** [comment]
Hello!
Thank you very much for this bug report. This is a problem which was introduced in 3.11 as part of some additional query optimizations. In the meantime, it has been fixed. The PR is here: https://github.com/arangodb/arangodb/pull/19571 There is also a 3.11 backport here: https://github.com/arangodb/arangodb/pull/19569
So this will be fixed in 3.11.3 when it comes out.
Best regards,
  Max.
